src/sema/ast.rs

```diff
@@ -187,6 +197,17 @@ impl Function {
         u32::from_le_bytes([res[0], res[1], res[2], res[3]])
     }
 
+    /// Generate selector with this function name for substrate
+    pub fn selector_blake2(&self) -> u32 {
+        let name= if self.name.is_empty() {
```

Comment:
`cargo fmt ` does not like this.

There can also be multiple constructors and they must have different selectors.
Solidity permits functions to be overloaded, so functions with the same name must have different selectors.

This is presumably why the tests are failing.

Reply:
Hi, @seanyoung , is there any way to adapt the selector of substrate ink! ? Because if we want to use the ink! contract on the substrate to call solang contract, the selector must be consistent.